### PR TITLE
added weighted sum

### PR DESCRIPTION
**Pull request by TezRomacH**
## Description

- Weighted sum fo `CriterionAggregatorCallback`
- Update tutorial

## Related Issue

N/A

## Type of Change

- [x] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves an existing feature)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have read the [Code of Conduct](https://github.com/catalyst-team/catalyst/blob/master/CODE_OF_CONDUCT.md) document.
- [x] I have read the [Contributing](https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md) guide.
- [x] I have checked the code-style using `make check-style`.
- [x] I have written the docstring in Google format for all the methods and classes that I used.
- [x] I have checked the docs using `make check-docs`.

### CODE REVIEW

**Comment by BloodAxe** [comment]
I think this can be already achieved by using existing `multiplier` property of `CriterionCallback`. One can set arbitrary multipliers to each criterion callback and achieve same effect of "mean", "sum" or "weighted sum" aggregation. And this would not add unnecessary complexity to `CriterionAggregationCallback`.

**Comment by TezRomacH** [comment]
Yes, this is the right remark, the same functionality can be implemented via scale.

But, as we have noticed in ODS, in this case the metric in the terminal and the tensorboard will also scales. And this is not good, for example, if you multiply by a small factor like 0.0001

File: catalyst/utils/dataset.py [hunk 1]
```diff
@@ -53,9 +53,8 @@ def create_dataset(
     return dataset
 
 
-def split_dataset(
-    dataset: pd.DataFrame, **train_test_split_args
-) -> Tuple[DictDataset, DictDataset]:
+def split_dataset(dataset: pd.DataFrame,
```
Comment (TezRomacH) [inline catalyst/utils/dataset.py]: weird, but it's from flake8

File: catalyst/utils/dataset.py [hunk 2]
```diff
@@ -316,9 +317,8 @@ def balance_classes(
     return balanced_df
 
 
-def prepare_dataset_labeling(
-    dataframe: pd.DataFrame, class_column: str
-) -> Dict[str, int]:
+def prepare_dataset_labeling(dataframe: pd.DataFrame,
```
Comment (TezRomacH) [inline catalyst/utils/dataset.py]: also from flake8

File: .travis.yml [hunk 3]
```diff
@@ -38,7 +38,7 @@ jobs:
     - stage: Tests
       install:
         - *requirements_all
-        - pip install flake8 flake8-quotes yapf isort
+        - pip install -U flake8 flake8-quotes yapf isort
```
Comment (Scitator) [inline .travis.yml]: what do you think about dumping the test packages versions for both local and CI tests?

File: catalyst/utils/dataset.py (hunk 1 above)
Comment (Scitator) [reply catalyst/utils/dataset.py]: looks like flake8 version change, we need to fix this... issue :)

**Comment by Scitator** [comment]
The PR looks good, I will try to dump the tests dependencies for CI correctness.

**Comment by Scitator** [comment]
@TezRomacH  could you please merge with master? we have fixed the CI tests issue

File: catalyst/utils/dataset.py (hunk 1 above)
Comment (TezRomacH) [reply catalyst/utils/dataset.py]: done

**Comment by TezRomacH** [comment]
updated

**Comment by lightforever** [comment]
@TezRomacH , please ensure you have the libraries in the list: https://github.com/catalyst-team/catalyst/blob/master/requirements/requirements-dev.txt

The versions must be the same. Especially, yapf must have version 0.28.0. The build works with this configuration.
